runtime/lua/vim/shared.lua

```diff
@@ -793,6 +793,7 @@ function vim._list_remove(t, first, last)
   end
 end
 
+local _trim_pattern = vim.re.compile('%s* {(%s* %S+)*}')
```

Comment:
`shared.lua` is special because the test runner uses it with a limited Nvim (`nvim -ll`) which doesn't have some modules like `vim.re`.

    /Users/runner/work/neovim/neovim/runtime/lua/vim/shared.lua:796: attempt to index field 're' (a nil value)

to workaround this, try:

```suggestion
local _trim_pattern = vim.re and vim.re.compile('%s* {(%s* %S+)*}') or nil
```

then below,

    return _trim_pattern and _trim_pattern:match(s) or (s:match('^%s*(.*%S)') or '')

alternatively, could make the `vim.re` module available by updating `src/nvim/CMakeLists.txt` (I think?). @bfredl @zeertzjq any reason `vim.re` can't be available for `nvim -ll` ?

Reply:
@justinmk Maybe then using lua loops with string.byte which is proposed by @glepnir is a better solution? And it is as fast if not faster than lpeg?

Reply:
If it's not a lot of code, then sure.

Reply:
The ["conditional gsub"](https://github.com/neovim/neovim/pull/34122#issuecomment-2901559179) solution is not a lot of code. It also outperforms lpeg approach in the other edge case of the very wide string that doesn't need trimming.

The downside of the `vim.re` approach is that it will always source `require('vim.re')` on startup, which has not quite negligible effect. On my machine a `nvim --clean -u init-vimre.lua` with 'init-vimre.lua' containing single line `local _trim_pattern = vim.re.compile('%s* {(%s* %S+)*}')` adds about 0.7ms to startup compared to overall about 10.3 ms of "Embedded" part of startup.